src/reducers/ast.js

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+/**
+ * UI reducer
+ * @module reducers/ui
+ */
+
+import makeRecord from "../utils/makeRecord";
+import constants from "../constants";
+import type { Action } from "../actions/types";
+import type { Record } from "../utils/makeRecord";
+import * as I from "immutable";
+
+export type ASTState = {};
+
+export const State = () =>
+  I.Map({
+    symbols: I.Map()
+  });
+
+function update(
+  state: Record<ASTState> = State(),
+  action: Action
+): Record<ASTState> {
+  switch (action.type) {
+    case "SET_SYMBOLS": {
+      const { source, symbols } = action;
+      return state.setIn(["symbols", source.id], symbols);
```

Comment:
@bomsy I like this solution because the reducer is an I.Map, the symbol field is an I.Map, but the symbols are just objs... maybe it's not a big deal bc people could still mess with them, but it feels better lol